setuptools/config/expand.py

```diff
@@ -203,7 +203,9 @@ def _load_spec(spec: ModuleSpec, module_name: str) -> ModuleType:
         return sys.modules[name]
     module = importlib.util.module_from_spec(spec)
     sys.modules[name] = module  # cache (it also ensures `==` works on loaded items)
-    spec.loader.exec_module(module)  # type: ignore
+    if spec.loader is None:
+        raise AttributeError(f"spec {spec} is missing a loader")
```

Comment:
or this if you think it should really never happen and want to remove the check in optimized mode

```suggestion
    assert spec.loader is not None
```

Reply:
I prefer the suggested line. It's more succinct and seems to carry the same expectation.